File: Lab1/advanced/Gausian.cpp
#include "TH1.h"
#include "TRandom3.h"
#include "TCanvas.h"
#include "TFile.h"
#include "TStyle.h"
#include <iostream>

int main(int argc, char const *argv[]) {
  // Create output file
  TFile *out = new TFile("out.root", "RECREATE");
  TCanvas *c1 = new TCanvas("c1", "Root Example", 200, 10, 700, 500);
  // Varibles for the gaussian
  int num = 100000;
  double mean = 5.0;
  double sigma = 1.0;
  double thrw = 0;

  if (argc > 1) {
    num = atoi(argv[1]);
  } else if (argc > 2) {
    num = atoi(argv[1]);
    mean = atof(argv[2]);
  } else if (argc > 3) {
    num = atoi(argv[1]);
    mean = atof(argv[2]);
    sigma = atof(argv[3]);
  }

  // Create random generator
  TRandom3 gen;

  // Create histogram to fill
  TH1F *hist = new TH1F("hist", "Gaussian", 100, 0.0, 10.0);

  for (int i = 0; i < num; i++) {
    // Throw a normal(gaussian) distrobution
    thrw = gen.Gaus(mean, sigma);
    // Fill in histogram
    hist->Fill(thrw);
  }

  // Fit gaussian to the histogram
  hist->Fit("gaus");
  // Save histogram to output file
  out->cd();
  out->Write();

  return 0;
}

File: Lab1/Gausian.py
#!/usr/bin/env python
from ROOT import TH1D, TCanvas, TF1
from ROOT import gStyle
from numpy.random import normal

# Varibles for the gaussian
num = 100000
mean = 5.0
sigma = 1.0

# Define the histogram to fill
hist = TH1D("hist", "Gaussian", 100, 0.0, 10.0)
# Define the gui to display the graph and set the display options
c1 = TCanvas('c1', 'Python Example', 200, 10, 700, 500)
gStyle.SetOptStat(111)
gStyle.SetOptFit(111)

# Thrown a normal(gaussian) distobution using numpy
throw = normal(loc=mean, scale=sigma, size=num)
# Loop over the thrown event and fill in histogram
#for t in throw:
#    hist.Fill(t)
# Same as above but in one line
[hist.Fill(t) for t in throw]

# Enter the canvas to draw
c1.cd()
# Draw histogram in gui
hist.Draw()
# Fit gaussian to the histogram
hist.Fit("gaus")

#Do this so it leaves the graph on the screen
raw_input("Press Enter to continue...")
